aws/resource_aws_cloudwatch_event_rule.go

```diff
@@ -121,6 +122,10 @@ func resourceAwsCloudWatchEventRuleCreate(d *schema.ResourceData, meta interface
 
 	log.Printf("[INFO] CloudWatch Event Rule %q created", *out.RuleArn)
 
+	if err := setTagsCloudWatchEvents(conn, d, aws.StringValue(out.RuleArn)); err != nil {
```

Comment:
For future enhancement, it appears `PutRule` supports a `Tags` parameter directly: https://docs.aws.amazon.com/sdk-for-go/api/service/cloudwatchevents/#PutRuleInput

Instead of setting the Tags via `TagResource` we may be able to simplify with something like:

```go
Tags: tagsFromMapCloudWatchEvents(d.Get("tags").(map[string]interface{})),
```

Which (at least on creation) may be a requirement in some environments with IAM-based restrictions based on tags.

The current implementation is passing acceptance testing though, so for now we can get this in as-is.